armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -225,8 +225,33 @@ class can then use to modify the isotopics as necessary.
             constructedObject = components.factory(shape, [], kwargs)
             _setComponentFlags(constructedObject, self.flags, blueprint)
             insertDepletableNuclideKeys(constructedObject, blueprint)
+
+        # set the custom density for non-custom material components after construction
+        self.setCustomDensity(constructedObject, blueprint)
+
         return constructedObject
 
+    def setCustomDensity(self, constructedComponent, blueprint):
+        """Apply a custom density to a material with custom isotopics but not a 'custom material'."""
+        if self.isotopics is not None:
+            # must nest because one can't check for attribues on a NoneType
+            if blueprint.customIsotopics[self.isotopics].density is not None:
+                mat = materials.resolveMaterialClassByName(self.material)()
+                if not isinstance(mat, materials.Custom):
+                    densityRatio = (
+                        blueprint.customIsotopics[self.isotopics].density
```

Comment:
do we always have a density? i thought some custom isotopic input options maybe didnt specify density. maybe they all eventually resolve it though

Reply:
No, the custom isotopics entry can be specified without a density, which is why line 238 checks for it. But if `customIsotopics` exists then it's attributes are initialized to `None`